### ISSUE TITLE
Installation randomly fails

### ISSUE THREAD

**Issue by SiM** [YunoHost-Apps/dolibarr_ynh#10]
For some unknown reason, the installation randomly fails. 
It seems the problem appears at "step5" of the install process, regarding the logs.

**Log of a succeed intall**
```
<div class="divlogoinstall" style="text-align:center"><img class="imglogoinstall" src="../theme/dolibarr_logo.png" alt="Dolibarr logo"><br>6.0.4</div><br><span class="titre">Installation ou mise &agrave; jour de Dolibarr - Fin de l&#039;installation</span>                                                                                                                                          
<form name="forminstall" style="width: 100%" action="step5.php" method="POST">                                                                                                                        
<input type="hidden" name="testpost" value="ok">                                                                                                      
<input type="hidden" name="action" value="set">                                                    
<table class="main" width="100%"><tr><td>                                                                                                                                                             
<table class="main-inside" width="100%"><tr><td>
<br>Cr&eacute;ation du compte administrateur Dolibarr &#039;<b>tera</b>&#039; r&eacute;ussie.<br>Activation du module modLdap<br>Votre syst&egrave;me est maintenant install&eacute;.<br><br>Vous devez maintenant configurer Dolibarr selon vos besoins (Choix de l&#039;apparence, des fonctionnalit&eacute;s, etc.). Pour cela, cliquez sur le lien ci-dessous&nbsp;:<br><br><div class="center"><a href="../admin/index.php?mainmenu=home&leftmenu=setup&username=tera">Acc&eacute;der &agrave; Dolibarr (espace de configuration)</a></div></td></tr></table></td></tr></table>
<input type="hidden" name="selectlang" value="fr_FR"></form>

<!-- Start of log output
[...]
End of log output -->                                                                                                                                                                                 
                                                                                                                                                                                                      
</body>                                                                                                                                                                                               
</html>                                                                                                                                                                                               
+ mysql -u dolibarr -p5D3tieBMiVQc dolibarr                                                                                                                                                           
+ mysql -u dolibarr -p5D3tieBMiVQc dolibarr                                                                                                                                                           
+ sudo sudo -u www-data php /var/www/dolibarr/scripts/user/sync_users_ldap2dolibarr.php commitiferror --server=localhost -y                                                                           
***** sync_users_ldap2dolibarr.php (6.0.4) pid=8009 *****                                                                                                                                             
Mails sending disabled (useless in batch mode)

----- Synchronize all records from LDAP database:
host=localhost
port=389
login=
pass=
DN to extract=ou=users,dc=yunohost,dc=org
Filter=(&(objectClass=posixAccount))
----- To Dolibarr database:
type=mysqli
host=localhost
port=3306
login=dolibarr
database=dolibarr
----- Options:
commitiferror=1
excludeuser=
Mapped LDAP fields=uid,cn,sn,givenName,telephonenumber,facsimiletelephonenumber,mobile,mail

UserCreate # allan: login=allan, fullname=Allan --> Created user id=2 login=allan
[...]
UserCreate # yves: login=yves, fullname=Yves --> Created user id=33 login=yves

Pas d'erreur, on valide

+ '[' 0 = 1 ']'
+ ynh_app_setting_delete dolibarr unprotected_uris
+ sudo yunohost app setting -d dolibarr unprotected_uris --quiet
+ sudo yunohost app ssowatconf
La configuration de SSOwat a été générée
+ sudo sed -i 's@$dolibarr_main_authentication='\''dolibarr'\'';@$dolibarr_main_authentication='\''http'\'';@g' /var/www/dolibarr/htdocs/conf/conf.php
+ sed -i s@YNH_APP@dolibarr@g ../hooks/post_user_create
+ sed -i s@YNH_MEMBER@0@g ../hooks/post_user_create
+ ynh_exit_properly
+ local exit_code=0
+ '[' 0 -eq 0 ']'
+ exit 0
Succès ! La configuration de SSOwat a été générée
Succès ! Installation terminée
```

**Log of a failed install**
```
<div class="divlogoinstall" style="text-align:center"><img class="imglogoinstall" src="../theme/dolibarr_logo.png" alt="Dolibarr logo"><br>6.0.4</div><br><span class="titre">Installation ou mise &agrave; jour de Dolibarr - Fin de l&#039;installation</span>
<form name="forminstall" style="width: 100%" action="step5.php" method="POST">
<input type="hidden" name="testpost" value="ok">
<input type="hidden" name="action" value="set">
<table class="main" width="100%"><tr><td>
<table class="main-inside" width="100%"><tr><td>
<br>
+ mysql -u dolibarr -pHr6pXJ5Vu6qG dolibarr
+ mysql -u dolibarr -pHr6pXJ5Vu6qG dolibarr
+ sudo sudo -u www-data php /var/www/dolibarr/scripts/user/sync_users_ldap2dolibarr.php commitiferror --server=localhost -y
***** sync_users_ldap2dolibarr.php (6.0.4) pid=5451 *****
Mails sending disabled (useless in batch mode)

----- Synchronize all records from LDAP database:
host=localhost
port=389
login=
pass=
DN to extract=ou=users,dc=yunohost,dc=org
Filter=(&(objectClass=posixAccount))
----- To Dolibarr database:
type=mysqli
host=localhost
port=3306
login=dolibarr
database=dolibarr
----- Options:
commitiferror=1
excludeuser=
Mapped LDAP fields=uid,cn,sn,givenName,mail

UserCreate # allan: login=allan, fullname=Allan --> -1 Table 'dolibarr.ynh_user' doesn't exist
[...]
UserCreate # yves: login=yves, fullname=Yves --> -1 Table 'dolibarr.ynh_user' doesn't exist

Errors found but we validate despite this

+ ynh_exit_properly
+ local exit_code=33
+ '[' 33 -eq 0 ']'
+ trap '' EXIT
+ set +eu
+ echo -e '!!\n  dolibarr'\''s script has encountered an error. Its execution was cancelled.\n!!'
+ type -t ynh_clean_setup
+ ynh_die
+ echo ''
+ exit 1

!!
dolibarr's script has encountered an error. Its execution was cancelled.
!!

Exécution du script « /var/cache/yunohost/from_file/dolibarr_ynh-master/scripts/remove »...
+ source _common.sh
+ source /usr/share/yunohost/helpers
+++ run-parts --list /usr/share/yunohost/helpers.d
++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'
++ '[' -r /usr/share/yunohost/helpers.d/backend ']'
++ . /usr/share/yunohost/helpers.d/backend
++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'
++ '[' -r /usr/share/yunohost/helpers.d/filesystem ']'
++ . /usr/share/yunohost/helpers.d/filesystem
+++ CAN_BIND=1
++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'
++ '[' -r /usr/share/yunohost/helpers.d/ip ']'
++ . /usr/share/yunohost/helpers.d/ip
++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'
++ '[' -r /usr/share/yunohost/helpers.d/mysql ']'
++ . /usr/share/yunohost/helpers.d/mysql
+++ MYSQL_ROOT_PWD_FILE=/etc/yunohost/mysql
++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'
++ '[' -r /usr/share/yunohost/helpers.d/network ']'
++ . /usr/share/yunohost/helpers.d/network
++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'
++ '[' -r /usr/share/yunohost/helpers.d/package ']'
++ . /usr/share/yunohost/helpers.d/package
++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'
++ '[' -r /usr/share/yunohost/helpers.d/print ']'
++ . /usr/share/yunohost/helpers.d/print
++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'
++ '[' -r /usr/share/yunohost/helpers.d/setting ']'
++ . /usr/share/yunohost/helpers.d/setting
++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'
++ '[' -r /usr/share/yunohost/helpers.d/string ']'
++ . /usr/share/yunohost/helpers.d/string
++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'
++ '[' -r /usr/share/yunohost/helpers.d/system ']'
++ . /usr/share/yunohost/helpers.d/system
++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'
++ '[' -r /usr/share/yunohost/helpers.d/user ']'
++ . /usr/share/yunohost/helpers.d/user
++ for helper in '$(run-parts --list /usr/share/yunohost/helpers.d 2>/dev/null)'
++ '[' -r /usr/share/yunohost/helpers.d/utils ']'
++ . /usr/share/yunohost/helpers.d/utils
+ app=dolibarr
++ ynh_app_setting_get dolibarr domain
++ sudo yunohost app setting dolibarr domain --output-as plain --quiet
yunohost.app.setting
+ domain=agora-dev.tera.coop
+ sudo rm -rf /var/www/dolibarr
+ sudo rm -f /etc/nginx/conf.d/agora-dev.tera.coop.d/dolibarr.conf
+ sudo rm -f /etc/php5/fpm/pool.d/dolibarr.conf
+ sudo service php5-fpm reload
+ dbname=dolibarr
+ dbuser=dolibarr
+ ynh_mysql_drop_db dolibarr
+ ynh_mysql_execute_as_root 'DROP DATABASE dolibarr;'
++ sudo cat /etc/yunohost/mysql
+ ynh_mysql_connect_as root 2OqvCGBZSW ''
+ mysql -u root --password=2OqvCGBZSW -B ''
+ ynh_mysql_drop_user dolibarr
+ ynh_mysql_execute_as_root 'DROP USER '\''dolibarr'\''@'\''localhost'\'';'
++ sudo cat /etc/yunohost/mysql
+ ynh_mysql_connect_as root 2OqvCGBZSW ''
+ mysql -u root --password=2OqvCGBZSW -B ''
+ sudo service nginx reload
La configuration de SSOwat a été générée
Échec de l'installation

```

```
# yunohost --version
yunohost: 
  repo: stable
  version: 2.7.10
yunohost-admin: 
  repo: stable
  version: 2.7.7
moulinette: 
  repo: stable
  version: 2.7.7
ssowat: 
  repo: stable
  version: 2.7.7

```

**Comment by SiM** [comment]
I try to install again Dolibarr (Jessie, Yunohost 2.7.12) and now the install fails at every attempt.
The error is the same : _Table 'dolibarr.ynh_user' doesn't exist_. 
Sometimes I have the same result that the log file I pasted in my first message, sometimes I have a different log : 

```
[...]
<form name="forminstall" style="width: 100%" action="step5.php" method="POST">
<input type="hidden" name="testpost" value="ok">
<input type="hidden" name="action" value="set">
<table class="main" width="100%"><tr><td>                                                     
<table class="main-inside" width="100%"><tr><td>                                                                           
<br>ERROR in activating module file=modUser.class.php<br><div class="error">Echec de la cr&eacute;ation du compte administrateur Dolibarr Table 'dolibarr.ynh$
user' doesn't exist</div><br><br>Dolibarr a d&eacute;tect&eacute; une erreur technique.<br>    
Voici les informations qui pourront aider au diagnostic:<br>                                                                                               
<b>Date:</b> 20180601173141<br>                                                                                             
<b>Dolibarr:</b> 6.0.4<br>                                                                                           
<b>PHP:</b> 5.6.33-0+deb8u1<br>                           
<b>Serveur:</b> nginx/1.6.2<br>
<b>OS:</b> Linux *** x86_64<br>
<b>UserAgent:</b> curl/7.38.0<br>
<br>   
<b>Url sollicit&eacute;e:</b> /dolibarr/install/step5.php<br>
<b>Referer:</b> <br>
<b>Gestionnaire de menu:</b> <br>
<br>
<b>Message:</b> step5.php: unknown choice of action<br>
</td></tr></table>
</td></tr></table>
<input type="hidden" name="selectlang" value="fr_FR"></form>

<!-- Start of log output
2018-06-01 17:31:41 --- step5: entering step5.php page<br>
2018-06-01 17:31:41 DoliDBMysqli::connect host=localhost, port=3306, login=dolibarr, passwd=--hidden--, name=<br>
2018-06-01 17:31:41 DoliDBMysqli::select_db database=dolibarr<br>
2018-06-01 17:31:41 step5: load module user ../core/modules/modUser.class.php<br>
2018-06-01 17:31:41 sql=SELECT name, label, fieldrequired FROM ynh_extrafields WHERE elementtype = 'user' AND entity = 1<br>
2018-06-01 17:31:41 DoliDBMysqli::query SQL Error message: DB_ERROR_NOSUCHTABLE Table 'dolibarr.ynh_extrafields' doesn't exist<br>
2018-06-01 17:31:41 BEGIN Transaction<br>
2018-06-01 17:31:41 modUser::_unactive<br>
2018-06-01 17:31:41 sql=DELETE FROM ynh_const WHERE name = 'MAIN_MODULE_USER' AND entity IN (0, 0)<br>
2018-06-01 17:31:41 DoliDBMysqli::query SQL Error message: DB_ERROR_NOSUCHTABLE Table 'dolibarr.ynh_const' doesn't exist<br>
2018-06-01 17:31:41 modUser::delete_tabs<br>
2018-06-01 17:31:41 sql=DELETE FROM ynh_const WHERE name like 'MAIN_MODULE_USER_TABS_%' AND entity = 1<br>
2018-06-01 17:31:41 DoliDBMysqli::query SQL Error message: DB_ERROR_NOSUCHTABLE Table 'dolibarr.ynh_const' doesn't exist<br>
2018-06-01 17:31:41 ROLLBACK Transaction<br>
2018-06-01 17:31:41 BEGIN Transaction<br>
2018-06-01 17:31:41 modUser::_active<br>
2018-06-01 17:31:41 sql=DELETE FROM ynh_const WHERE name = 'MAIN_MODULE_USER' AND entity IN (0, 0)<br>
2018-06-01 17:31:41 DoliDBMysqli::query SQL Error message: DB_ERROR_NOSUCHTABLE Table 'dolibarr.ynh_const' doesn't exist<br>
2018-06-01 17:31:41 modUser::_active<br>
2018-06-01 17:31:41 sql=INSERT INTO ynh_const (name, value, visible, entity, note) VALUES ('MAIN_MODULE_USER', '1', 0, 0, '{\"authorid\":0,\"ip\":\"::1\"}')<br>
2018-06-01 17:31:41 DoliDBMysqli::query SQL Error message: DB_ERROR_NOSUCHTABLE Table 'dolibarr.ynh_const' doesn't exist<br>
2018-06-01 17:31:41 ROLLBACK Transaction<br>
2018-06-01 17:31:41 Conf::setValues<br>
2018-06-01 17:31:41 sql=SELECT name as name, value as value, entity FROM ynh_const WHERE entity IN (0,1) ORDER BY entity<br>
2018-06-01 17:31:41 DoliDBMysqli::query SQL Error message: DB_ERROR_NOSUCHTABLE Table 'dolibarr.ynh_const' doesn't exist<br>
2018-06-01 17:31:41 sql=SELECT u.rowid, u.pass, u.pass_crypted FROM ynh_user as u<br>
2018-06-01 17:31:41 DoliDBMysqli::query SQL Error message: DB_ERROR_NOSUCHTABLE Table 'dolibarr.ynh_user' doesn't exist<br>
2018-06-01 17:31:41 User::create login=test, user=0<br>
2018-06-01 17:31:41 BEGIN Transaction<br>
2018-06-01 17:31:41 User::create<br>
2018-06-01 17:31:41 sql=SELECT login FROM ynh_user WHERE login ='test' AND entity IN (0,1)<br>
2018-06-01 17:31:41 DoliDBMysqli::query SQL Error message: DB_ERROR_NOSUCHTABLE Table 'dolibarr.ynh_user' doesn't exist<br>
2018-06-01 17:31:41 ROLLBACK Transaction<br>
2018-06-01 17:31:41 step5: FailedToCreateAdminLogin Table 'dolibarr.ynh_user' doesn't exist<br>
2018-06-01 17:31:41 sql=INSERT INTO ynh_const(name,value,type,visible,note,entity) VALUES ('MAIN_LANG_DEFAULT','fr_FR','chaine',0,'Default language',1)<br>
2018-06-01 17:31:41 DoliDBMysqli::query SQL Error message: DB_ERROR_NOSUCHTABLE Table 'dolibarr.ynh_const' doesn't exist<br>
2018-06-01 17:31:41 Error url=/dolibarr/install/step5.php, query_string=, msg=step5.php: unknown choice of action<br>
2018-06-01 17:31:41 --- step5: Dolibarr setup finished<br>
End of log output -->
[...]
```

But in both cases, I noticed that before theses errors, there is a timeout after step2 : 
```
+ ynh_local_curl /install/step2.php testpost=ok action=set dolibarr_main_db_character_set=latin1 dolibarr_main_db_collation=latin1_swedish_ci selectlang=fr_FR
+ local full_page_url=https://localhost/dolibarr/install/step2.php
+ local POST_data=                                                                                                                                            
+ local arg=                                                                                                                
+ for arg in '"${@:2}"'                                      
+ POST_data='testpost=ok&'                                                                                                                                    
+ for arg in '"${@:2}"'        
+ POST_data='testpost=ok&action=set&'                                                                                                                         
+ for arg in '"${@:2}"'                                                                                         
+ POST_data='testpost=ok&action=set&dolibarr_main_db_character_set=latin1&'   
+ for arg in '"${@:2}"'                                                                                                     
+ POST_data='testpost=ok&action=set&dolibarr_main_db_character_set=latin1&dolibarr_main_db_collation=latin1_swedish_ci&'    
+ for arg in '"${@:2}"'                                                                                                                                       
+ POST_data='testpost=ok&action=set&dolibarr_main_db_character_set=latin1&dolibarr_main_db_collation=latin1_swedish_ci&selectlang=fr_FR&'
+ '[' -n 'testpost=ok&action=set&dolibarr_main_db_character_set=latin1&dolibarr_main_db_collation=latin1_swedish_ci&selectlang=fr_FR&' ']'                    
+ POST_data='--data testpost=ok&action=set&dolibarr_main_db_character_set=latin1&dolibarr_main_db_collation=latin1_swedish_ci&selectlang=fr_FR'               
+ curl --silent --show-error -kL -H 'Host: test.tera.coop' --resolve test.tera.coop:443:127.0.0.1 --data 'testpost=ok&action=set&dolibarr_main_db_character_se
t=latin1&dolibarr_main_db_collation=latin1_swedish_ci&selectlang=fr_FR' https://localhost/dolibarr/install/step2.php                                          
                                                     
<html>                                                                                                                                                        
<head><title>504 Gateway Time-out</title><script type="text/javascript" src="/ynhpanel.js"></script></head>                                                   
<body bgcolor="white">                                                                                                                                        
<center><h1>504 Gateway Time-out</h1></center>                                                                                                                
<hr><center>nginx</center>                                                                                                                                    
</body>                                                                                                                            
</html>                          
+ ynh_local_curl /install/step4.php testpost=ok action=set selectlang=fr_FR
+ local full_page_url=https://localhost/dolibarr/install/step4.php    
```

Maybe that's why the database isn't filled out properly ?

I don't know how to investigate / test further.

**Comment by mastereur** [comment]
I think that this issue is no longer relevant since the last message and the reworking (after July 05, 2019).

**Comment by mastereur** [comment]
closed due to inactivity
